### DISCUSSION TITLE
DataflowStartFlexTemplateOperator could not go through due to KeyError 'Job' from Response and validationOnly flag is true in JSON payload

### DISCUSSION THREAD

**Discussion by myang-clgx** [apache/airflow#24274] [General]
### Apache Airflow Provider(s)

google

### Versions of Apache Airflow Providers

apache-airflow-providers-apache-beam 3.4.0
apache-airflow-providers-ftp         2.1.2
apache-airflow-providers-google      7.0.0
apache-airflow-providers-http        2.1.2
apache-airflow-providers-imap        2.2.3
apache-airflow-providers-sqlite      2.1.3

### Apache Airflow version

2.3.1

### Operating System

MAC OSX 12.4

### Deployment

Other

### Deployment details

Just Local Deployment without Docker

### What happened

Try to run a Dag with Dataflow Flex template operator like below,

```
@task
def get_main_config():
   ...
   return config

@task
def get_dataflow_node_config(main_json_config: dict):
   ...
   return config

def custom_dataflow_etl(**kwargs):
    fetch_main_config = get_main_config()
    fetch_dataflow_config_dic = get_dataflow_node_config(fetch_main_config)
    process_dataflow = DataflowStartFlexTemplateOperator(
            task_id="start_dataflow_flex_template",
            body=fetch_dataflow_config_dic, do_xcom_push=True, location=DATAFLOW_LOCATION)
    fetch_dataflow_config >> process_dataflow

custom_dataflow = custom_dataflow_etl()
```

It gives Exception when do DAG RUN.

```
[2022-06-03, 13:34:29 UTC] {taskinstance.py:1890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/Users/myang/opt/miniconda3/envs/airflow-2_3_1/lib/python3.9/site-packages/airflow/providers/google/cloud/operators/dataflow.py", line 773, in execute
    job = self.hook.start_flex_template(
  File "/Users/myang/opt/miniconda3/envs/airflow-2_3_1/lib/python3.9/site-packages/airflow/providers/google/common/hooks/base_google.py", line 439, in inner_wrapper
    return func(self, *args, **kwargs)
  File "/Users/myang/opt/miniconda3/envs/airflow-2_3_1/lib/python3.9/site-packages/airflow/providers/google/cloud/hooks/dataflow.py", line 754, in start_flex_template
    job = response["job"]
KeyError: 'job'
```

### What you think should happen instead

It will go through because I have tried to use Google Cloud CLI to test out the DataflowFlexTemplate with same test values on parameters and it executes.

### How to reproduce

Create a DAG and use Dataflextemplateoperator like below,

```python
@task
def get_main_config():
   ...
   return config

@task
def get_dataflow_node_config(main_json_config: dict):
   ...
   return config

def custom_dataflow_etl(**kwargs):
    fetch_main_config = get_main_config()
    fetch_dataflow_config_dic = get_dataflow_node_config(fetch_main_config)
    process_dataflow = DataflowStartFlexTemplateOperator(
            task_id="start_dataflow_flex_template",
            body=fetch_dataflow_config_dic, do_xcom_push=True, location=DATAFLOW_LOCATION)
    fetch_dataflow_config >> process_dataflow

custom_dataflow = custom_dataflow_etl()
```

### Anything else
Update on further investigation: 

This issue has nothing to do with versions of each pip libraries nor constraints.

I also check the line 754 under dataflow.py and the line is there for a while. I found the issue is behind causing this ERROR.
In DataflowStartFlexTemplateOperator usage on ```body=``` has a flag called, "validateOnly". If the flag is true, this will cause "KeyError 'Job'" from response obj. There should be a validation on line 753 and 754 either checking on response or the validationOnly flag in the request body before calling.
```
response = request.execute(num_retries=self.num_retries)
job = response["job"]
```
Lastly, probably need a friendly error msg.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by uranusjr** [comment]
In the future, please read GitHub’s Markdown guide and try to format the code blocks correctly. (I fixed them for you in this issue.)

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

**Reply by myang-clgx** [reply]
thank you

**Comment by potiuk** [comment]
Looking at the #24267  you simply installed airlfow without the right constraints. Please check it after installing it properly. I will convert it into discussion until you double-check it.

**Reply by myang-clgx** [reply]
@potiuk , I checked versions and compared with right constraints.  I used the AF official constraints file for installation.

I also check the line 754 under dataflow.py and the line is there for a while.  I found the issue is behind causing this ERROR.  
In DataflowStartFlexTemplateOperator input parameter on ```body=``` has a flag called, "validateOnly".  If the flag is true, this will cause "KeyError 'Job'" from response obj.  There should be a validation on line 753 and 754 either checking on response or the validationOnly flag before calling.   
```
response = request.execute(num_retries=self.num_retries)
job = response["job"]
```

Lastly, probably need a friendly error msg.  I still think this is a bug and need to be addressed.  Not the top priority.  I will edit the issue subject line for clarity.  Ofc, you can argue that it's an user usage error but I got nothing else to say if that's the case.

**Reply by potiuk** [reply]
Can you please provide a PR fixing it?  You seem to know exactly what the problem is.

**Reply by potiuk** [reply]
Airflow is created by > 2000 contributors like you, so when you found and error, know a fix, and have an environment that you can test it with (especially when it is one of 3000 operators that airflow has) seems like super cool contribution, that can help others to get similar problem solved.

**Reply by potiuk** [reply]
And yeah. Adding a helpful message is always welcome. Happy to review the PR and merge it

**Reply by myang-clgx** [reply]
Is it still under discussion or an issue?  I will think about the PR.  Thanks for the reply.

**Reply by potiuk** [reply]
PRs area always good. Feel free to open one. Discussions - by definitions - almost never end. Or they actually end when somoene (like you) create a PR out of it.
